src/X25519KeyAgreementKey2019.js

```diff
@@ -283,17 +230,9 @@ export class X25519KeyPair extends LDKeyPair {
    * @returns {String}
    */
   deriveSecret({publicKey}) {
-    const remotePubkeyBytes = base58Decode({
-      decode: base58.decode,
-      keyMaterial: publicKey.publicKeyBase58,
-      type: 'public'
-    });
+    const remotePubkeyBytes = decode(publicKey.publicKeyBase58);
 
-    const privateKeyBytes = base58Decode({
-      decode: base58.decode,
-      keyMaterial: this.privateKeyBase58,
-      type: 'private'
-    });
+    const privateKeyBytes = decode(this.privateKeyBase58);
 
     return nacl.scalarMult(privateKeyBytes, remotePubkeyBytes);
```

Comment:
See: https://github.com/digitalbazaar/minimal-cipher/blob/x25519/algorithms/x25519.js#L118-L139 <-- for node.js native X25519 implementation. We should add that option here.

Reply:
Got it. If you don't object, I'll just copy that `deriveSecret` function from `minimal-cipher` to here as well. Since I don't see a good way to factor it out / share it somehow.

Reply:
That's fine.

Reply:
There's keygen stuff in there too, if applicable here.

Reply:
@dlongley I added the native `deriveSecret` implementation.

Which native keygen code should I use? This [`deriveEphemeralKeyPair()`](https://github.com/digitalbazaar/minimal-cipher/blob/x25519/algorithms/x25519.js#L24) function?

Reply:
Yes.

Reply:
@dlongley Added native keygen, too. Should be all set.